androidcommon_lib/src/test/java/org/opendatakit/test/UrlUtilsTest.java

```diff
@@ -0,0 +1,143 @@
+package org.opendatakit.test;
+
+import org.junit.Test;
+import org.opendatakit.webkitserver.utilities.UrlUtils;
+
+import static org.junit.Assert.*;
+
+public class UrlUtilsTest {
+
+    private static final String EMPTY_STRING = "";
+    private static final String URL_SEGMENT1 = "test/";
+    private static final String FILE_PATH_PREFIX = "this/test/file/path/";
+    private static final String HTML_EXTENSION = ".html";
+    private static final String QUERY_PARAM_PREFIX = "?";
+    private static final String HASH_PREFIX = "#";
+
+    // File paths
+    private static final String FILE_PATH_WITH_HTML = FILE_PATH_PREFIX + HTML_EXTENSION;
+
+    // Query and hash parameter values
+    private static final String HASH_VALUE_FOO = "foo";
+    private static final String QUERY_VALUE_FOO_AND_BAR = "foo&bar=3";
+    private static final String QUERY_VALUE_BAR_AND_BAZ = "bar=3&baz=55";
+    private static final String QUERY_VALUE_FOO_BAR = "foo=bar";
+    private static final String QUERY_VALUE_BAR_BAZ = "bar=baz";
+
+    // Complete URL fragments with parameters
+    private static final String FILE_WITH_HASH = URL_SEGMENT1 + "file" + HASH_PREFIX + HASH_VALUE_FOO;
+    private static final String FILE_WITH_QUERY_PARAMS = "pretty/little/liar" + QUERY_PARAM_PREFIX + QUERY_VALUE_FOO_AND_BAR;
+    private static final String FILE_WITH_HASH_AND_QUERY = URL_SEGMENT1 + "test/test" + HTML_EXTENSION +
+            HASH_PREFIX + HASH_VALUE_FOO +
+            QUERY_PARAM_PREFIX + QUERY_VALUE_BAR_AND_BAZ;
+    private static final String FILE_WITHOUT_PARAMS = URL_SEGMENT1 + "test/test" + HTML_EXTENSION;
+    private static final String FILE_WITH_HASH_ONLY = URL_SEGMENT1 + "test" + HTML_EXTENSION +
+            HASH_PREFIX + HASH_VALUE_FOO;
+    private static final String FILE_WITH_QUERY = "this/is/a/file/that/i/like" + HTML_EXTENSION +
+            QUERY_PARAM_PREFIX + QUERY_VALUE_FOO_BAR;
+    private static final String FILE_WITH_BOTH = "foo/bar" + HTML_EXTENSION +
+            HASH_PREFIX + HASH_VALUE_FOO +
+            QUERY_PARAM_PREFIX + QUERY_VALUE_BAR_BAZ;
+
+    @Test
+    public void testNoHashOrParameters() {
+        assertRetrieveFileNameHelper(FILE_PATH_WITH_HTML, FILE_PATH_WITH_HTML);
+    }
+
+    @Test
+    public void testEmptyString() {
+        assertRetrieveFileNameHelper(EMPTY_STRING, EMPTY_STRING);
+    }
+
+    @Test
+    public void testOnlyHash() {
+        assertRetrieveFileNameHelper(URL_SEGMENT1 + "file", FILE_WITH_HASH);
+    }
+
+    @Test
+    public void testOnlyQueryParams() {
+        assertRetrieveFileNameHelper("pretty/little/liar", FILE_WITH_QUERY_PARAMS);
+    }
+
+    @Test
+    public void testHashAndQueryParams() {
+        assertRetrieveFileNameHelper(URL_SEGMENT1 + "test/test" + HTML_EXTENSION, FILE_WITH_HASH_AND_QUERY);
```

Comment:
One of your constants is almost the same... why not have the first part be a different constant that uses the first constant then adds a second constant

private static final String FILE_WITH_HASH_AND_QUERY = URL_SEGMENT1 + "test/test" + HTML_EXTENSION ...